src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -896,6 +907,49 @@ private void EnsureRavenEntityName(BlittableJsonReaderObject obj,string collecti
             }
             return true;
         }
+        
+        private bool TryResolveByLeader(
+            DocumentsOperationContext context,
+            string key)
+        {
+            var leader = _parent.ResolverLeader;
```

Comment:
Don't like the leader terminology here, we already have that concept in raft, and it is different.
ResolvingDatabase is a better term, I think